AWS Roles Anywhere: List Profiles API: return all profiles

marcoandredinis opened the pull request:
This PR changes the `listprofiles` endpoint to return all the profiles at once.

Pagination will be implemented client side.

Why do we need this change?
This endpoint is used to show the list of existing profiles when the user is setting up the profile sync using the AWS IAM Roles Anywhere integration.

Given that we have filters, it might happen that to list X profiles, we might need to do X api calls, which is not ideal.
Instead of asking for a paginated list of Profiles, the WebUI will ask for the whole list.

This list is limited to a max of 250 items as described in the AWS Roles Anywhere quotas:
https://docs.aws.amazon.com/rolesanywhere/latest/userguide/quotas.html

The new flow (after this PR):
```mermaid
sequenceDiagram
    WebUI->>+Proxy: HTTP API ListProfiles
    loop Until there's no profiles left
        Proxy->>+Auth: gRPC API ListProfiles
        Auth->>+AWS-API: HTTP Req ListProfiles
        AWS-API->>+Auth: Paginated List of Profiles
        Auth->>+Auth: Filter profiles
        Auth->>+Proxy: Paginated List of filtered Profiles
    end
    Proxy->>+WebUI: List of all filtered Profiles
```

zmb3 commented:
Can you update the PR description to describe _why_ we want to fetch everything instead of exposing a paginated API to the UI?

We're doing work right now to _remove_ APIs like this because they create issues at scale. Is it really necessary?

marcoandredinis commented on lib/web/ui/integration.go:
```diff
@@ -710,12 +710,8 @@ type AWSRolesAnywherePingResponse struct {
 }
 
-// AWSRolesAnywhereListProfilesRequest contains the list of Roles Anywhere Profiles.
+// AWSRolesAnywhereListProfilesRequest contains the list of all Roles Anywhere Profiles.
```
Fixed, thank you.

marcoandredinis committed:
AWS Roles Anywhere: List Profiles API: return all profiles

This PR changes the `listprofiles` endpoint to return all the profiles
at once.

This list is limited to a max of 250 items as described in the AWS Roles
Anywhere quotas:
https://docs.aws.amazon.com/rolesanywhere/latest/userguide/quotas.html

Pagination will be implemented client side.
